library/LuaTypes.cpp

```diff
@@ -706,6 +707,16 @@ static type_identity *find_primitive_field(lua_State *state, int field, const ch
  */
 static int meta_primitive_index(lua_State *state)
 {
+    const char *attr = lua_tostring(state, -1);
+    if (attr == std::string("ref_target")) {
+        const struct_field_info *field_info = get_object_ref_header(state, 1)->field_info;
+        if (field_info && field_info->extra && field_info->extra->ref_target) {
+            LookupInTable(state, field_info->extra->ref_target, &DFHACK_TYPEID_TABLE_TOKEN);
+        } else {
+            lua_pushnil(state);
+        }
+        return 1;
+    }
```

Comment:
This part in particular could use some cleaning up - one example is the special `whole` field for bitfields, defined [here](https://github.com/DFHack/dfhack/blob/0.47.04-beta1/library/LuaTypes.cpp#L1414) and implemented [here](https://github.com/DFHack/dfhack/blob/0.47.04-beta1/library/LuaTypes.cpp#L958-L965)